nimterop/cimport.nim

```diff
@@ -174,18 +174,33 @@ macro cSkipSymbol*(skips: varargs[string]): untyped =
 
 macro cPlugin*(body): untyped =
   ## When `cOverride() <cimport.html#cOverride.m,>`_ and `cSkipSymbol() <cimport.html#cSkipSymbol.m%2Cvarargs[string]>`_
-  ## are not adequate, the `cPlugin() <cimport.html#cPlugin.m,>`_ macro can be used to customize the generated Nim output.
-  ## The following callbacks are available at this time.
+  ## are not adequate, the `cPlugin() <cimport.html#cPlugin.m,>`_ macro can be used
+  ## to customize the generated Nim output. The following callbacks are available at
+  ## this time.
   ##
+  ## .. code-block:: nim
+  ##
+  ##     proc onSymbol(sym: var Symbol): Result {.exportc, dynlib.}
+  ##
+  ## `onSymbol()` can be used to handle symbol name modifications required due to invalid
+  ## characters like `_` or to rename duplicate types. It can also be used to remove prefixes
+  ## and suffixes. The symbol name and type is provided to the callback and the name can be
+  ## modified. Symbol types can be any of the following:
+  ## - `nskConst` for constants
+  ## - `nskType` for type identifiers, including primitive
+  ## - `nskParam` for param names
+  ## - `nskField` for struct field names
+  ## - `nskEnumField` for enum (field) names, though they are in the global namespace as `nskConst`
```

Comment:
> though they are in the global namespace as `nskConst`

i guess this relates to https://github.com/genotrance/nimterop/issues/72#issuecomment-457938613 ; we can discuss that separately

Reply:
They will still be consts though so will leave this in for now.